### DISCUSSION TITLE
Please add simple save preset functionality under Settings.

### DISCUSSION THREAD

**Discussion by jockisen** [lllyasviel/Fooocus#3267] [Ideas]
Request in title.

**Comment by ILikeToasters** [comment]
I am pretty sure this is on the eventual to do list but until then I do have a simple way to create presets from metadata: https://ai.kleebz.com/fooocus/presets/

**Reply by jockisen** [reply]
Yeah. Tried that one but it's faulty. Would be great with a Save preset button.

**Reply by ILikeToasters** [reply]
> Yeah. Tried that one but it's faulty. Would be great with a Save preset button.

How is it faulty? I can only fix it if I know what problem you may had. I tested it over 100 times after creating it so if you found an issue I would like to fix it.

**Comment by vitorfdl** [comment]
I would be happy with a just auto-save settings.
I have refreshed my page by mistake multiple times and lost all the prompt, loras, etc

**Reply by mashb1t** [reply]
This is intended as you can have multiple tabs opened with different settings and you can use Fooocus from multiple independent clients not being synched with the other sessions. All changes you do in your browser are client-side and not synched to the server (Fooocus), which is a feature in and of itself to prevent session overlap as it has been implemented in the early times.
If you have an idea how to optimize the current behavior keeping this in mind i'm happy to hear your proposal.

**Reply by Rogala** [reply]
When you create a picture and you like it, you can always take its parameters from the history of logs, this will be the saving of settings.

**Reply by vitorfdl** [reply]
Well, I think that can be solved simply by using Local/Session Storage in the browser/front-end. If a person is using multiple tabs, it's very likely he doesn't need this feature anyway and will not matter what is being saved in the local storage.

From an UX perspective, both types of user would have the benefit of opening a new tab with some last settings pre-filled (as you can easily come back to some default preset by using the dropdown menu).

Doing it on the browser will allow multiple clients with different saved settings.

**Reply by vitorfdl** [reply]
Oh, I understand the History Logs solution now (I just checked on this today). It makes sense why you guys were suggesting to use that. I also think it's a good support feature for saving prompts now.

I think it could benefit from some low-priority quality of life improvements:
- Adjust the History Log display to make it easier for new users to notice.
- Add a button in the History List row labeled "Open in Fooocus" for a quick way to get right into action.

**Comment by muratyuceer** [comment]
I need Save and Save As button too
